internal/integrations/envsec/envsec.go

```diff
@@ -3,15 +3,27 @@ package envsec
 import (
 	"bytes"
 	"encoding/json"
+	"os"
 	"os/exec"
+	"path/filepath"
 
 	"github.com/pkg/errors"
 	"go.jetpack.io/devbox/internal/boxcli/usererr"
 	"go.jetpack.io/devbox/internal/cmdutil"
+	"go.jetpack.io/devbox/internal/debug"
+	"go.jetpack.io/pkg/sandbox/runx"
 )
 
+var envCache map[string]string
+
 func Env(projectDir string) (map[string]string, error) {
 
+	defer debug.FunctionTimer().End()
+
+	if envCache != nil {
```

Comment:
How do you validate/invalidate this cache

Reply:
It never gets invalidated, it's lifetime is just the command. So if you do `devbox run` you only fetch it once.

Later on I think we can look into refreshing env vars for long running shells, but even then this cache is OK.